omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,145 @@
+          
+            if($scope.condition.status=='INACTIVE'){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = self.convertDate1($scope.condition.endDate);
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[1].value = self.convertDate2($scope.condition.onSetDate);  
+            self.showEndDate();
+        }
+
+    self.showEndDate = self.showEndDate || function() {
+        var inputs = document.getElementsByTagName('input');
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status=='INACTIVE'){
+            groups[2].style.visibility = "visible";
+            inputs[3].value = self.convertDate2($scope.condition.endDate);
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+    }
+
+    self.validateCondition = self.validateCondition || function () {
+            var concept ;
+            if($scope.concept.concept.uuid)
+            {
+                concept=$scope.concept.concept;
+            }
+            else{
+                concept=$scope.condition.concept;
+            }
+            if (concept !== null) {
+                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+            } else {
+                // noncoded diagnosis
+                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+            }
+
+            $scope.condition.onSetDate = self.getSelectedDate();
+            if($scope.condition.status=='INACTIVE')
+            {
+                $scope.condition.endDate = self.getEndDate();
+            }
+            self.saveCondition();
+        }
+
+    self.unselectStatus = self.unselectStatus || function () {
+            $scope.condition.status = null;
+        }
+
+    self.getSelectedDate = self.getSelectedDate || function () {
+            var datePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];
+            return datePicker.value;
+        }
+        
+    self.getEndDate = self.getEndDate || function() {
+           var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];
+           return endDatePicker.value;
+    }    
+
+    self.convertDate1 = self.convertDate1 || function(date_in_ms) {
+            var date = new Date(date_in_ms);
+            var y = date.getFullYear();
+            var m = date.getMonth() + 1;    
+            var d = date.getDate();    
+            m = (m < 10) ? '0' + m : m; 
+            d = (d < 10) ? '0' + d : d;
+            return [y, m, d].join('-');
+    }
+        
+    self.convertDate2 = self.convertDate2 || function(date_in_ms) {
+           var d = new Date(date_in_ms);
+           d = d.toUTCString();
+           d = d.split(" ").slice(1,4).join(" ");
```

Comment:
What is this code accomplishing?

Reply:
This code is to display the dates in the onset date and end date fields in the following format- 18 Jun 2019

Reply:
Is that obvious from the code?

Reply:
I will add a comment before the function

Reply:
What does convertForDateDisplay mean?

Reply:
Do you want me to name it to something else? If yes, Please suggest some name to this function.

Reply:
Have you looked at what the rest of the openmrs user interfaces use to format dates for display?

Reply:
I did, but I didn't find any implementation that could be of help in this context.

Reply:
Are these of help?
https://talk.openmrs.org/t/best-practice-for-switching-between-date-formats/10012

https://wiki.openmrs.org/display/docs/UI+Framework+Reference+Guide#UIFrameworkReferenceGuide-Dateformatting

https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/main/java/org/openmrs/ui/framework/UiUtils.java

https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/test/java/org/openmrs/ui/framework/FormatterImplTest.java